File: Programming 1/Pokemon Game/Pokemon.java
/** 
 * Pokemon.Java
 * Keith Monaghan
 * 2443563
 * 4-11-14
 * Pokemon object to be used in Colosseum.Java
 */

import java.util.Scanner;

public class Pokemon{
	
	//Pokemon's core variables
	private int hp, attackLevel, defenseLevel;
	private String name;
	private Dice d20, d6;
	
	//Constructor
	public Pokemon(){
		hp = 0;
		attackLevel = 0;
		defenseLevel = 0;
		
		name = "";
		
		d20 = new Dice(20);
		d6 = new Dice(6);
	}
	
	//Completes one attack
	public boolean attack(Pokemon opponent){
		System.out.println(this.getName() + " is attacking " + opponent.getName());
		
		int roll1 = this.d20.roll();
		int roll2 = this.d20.roll();
		
		System.out.println(this.name + " rolls an attack bonus of " + roll1);
		System.out.println(opponent.getName() + " rolls a defense bonus of " + roll2);
		
		if((this.attackLevel + roll1) > (opponent.getDefense() + roll2)){
			System.out.println("The attack hits dealing 3-D6 damage!");
			int roll3 = this.d6.roll();
			int roll4 = this.d6.roll();
			int roll5 = this.d6.roll();
			
			int total = roll3 + roll4 + roll5;
			
			opponent.setHP(opponent.getHP() - total);
			
			System.out.println("The rolls are are " + roll3 + ", " + roll4 + ", and " + roll5 + " totaling: " + total + " damage!");
			
			if(opponent.getHP() <= 0){
				return true;
			}
			else{
				System.out.println(opponent.getName() + " has " + opponent.getHP() + " hit points left");
				return false;				
			}
		}
		else{
			System.out.println("The attack missed!");
			return false;
		}
		
	}
	
	//Interacts with the user to initialize all of the variables.
	public void userBuild(){
		Scanner input = new Scanner(System.in);
		
		System.out.print("Please name your pokemon: ");
		this.name = input.nextLine();
		
		System.out.print("\nHow many hit points will it have? (1-50): ");
		this.hp = input.nextInt();
		while(this.hp < 1 || this.hp>50){
			System.out.print("\nSorry. Hit points must be between 1 and 50: ");
			this.hp = input.nextInt();
		}
		
		System.out.print("\nSplit fifty points between attack level and defense level\nEnter your attack level (1-49): ");
		this.attackLevel = input.nextInt();
		
		while(this.attackLevel < 1 || this.attackLevel > 49){
			System.out.print("\nSorry. The attack level must be between 1 and 49: ");
			this.attackLevel = input.nextInt();
		}
		
		System.out.print("\nEnter your defense level (1-" + (50-this.attackLevel) + "): ");
		this.defenseLevel = input.nextInt();
		
		while(this.defenseLevel < 1 || (this.attackLevel + this.defenseLevel) > 50){
			System.out.print("\nSorry. The defense level must be between 1 and " + (50-this.attackLevel) + ": ");
			this.defenseLevel = input.nextInt();
		}
	}
	
	
	//Get methods 
	public int getHP(){
		return hp;
	}
	
	public int getAttack(){
		return attackLevel;
	}
	
	public int getDefense(){
		return defenseLevel;
	}
	
	public String getName(){
		return name;
	}
	
	//Set methods. 
	//It seems like it would be much easier to just use a constructer for most of these. 
	public void setHP(int health){
		hp = health;
	}
	
	public void setAttack(int attack){
		attackLevel = attack;
	}
	
	public void setDefense(int defense){
		defenseLevel = defense;
	}
	
	public void setName(String NewName){
		name = NewName;
	}
}

File: Programming 2/Pokemon Game/Pokemon.cpp
#include "Pokemon.h"
#include "Dice.h"
#include <string>

Pokemon::Pokemon()
{
    this->hp = 0;
    this->atk = 0;
    this->d6 = Dice(6);
    this->d20 = Dice(20);
    this->def = 0;
    this->name = "";
}

int Pokemon::getAttack(){
    return(this->atk);
}

int Pokemon::getDefense(){
    return(this->def);
}

int Pokemon::getHP(){
    return(this->hp);
}

string Pokemon::getName(){
    return(this->name);
}

void Pokemon::setAttack(int newATK){
    this->atk = newATK;
}

void Pokemon::setDefense(int newDEF){
    this->def = newDEF;
}

void Pokemon::setHP(int newHP){
    this->hp = newHP;
}

void Pokemon::setName(string newName){
    this->name = newName;
}

void Pokemon::userBuild(){
    std::cout << "Please name your pokemon: ";
    std::cin >> this->name;
    do{
        std::cout << "\nHow many hit points will it have? (1-50): ";
        std::cin >> this->hp;
    }while(this->hp < 1 || this->hp > 50);

    std::cout << "Split fifty points between attack level and defense level" << endl << "Enter your attack level (1-49): ";
    std::cin >> this->atk;

    while(this->atk < 1 || this->atk > 49){
        std::cout << "\nSorry. The attack level must be between 1 and 49: ";
        std::cin >> this->atk;
    }

    std::cout << "Enter your defense level (1-" << (50-this->atk) << "): ";
    std::cin >> this->def;

    while(this->def < 1 || this->def > (50-this->atk)){
        std::cout << "Sorry. The defense level must be between 1 and " << (50-this->atk) << ": ";
        std::cin >> this->def;
    }
}

bool Pokemon::attack(Pokemon &opponent){
    int temp;
    int temp2;
    int temp3;

    std::cout << this->name << " is attacking " << opponent.getName() << endl;
    temp = this->d20.roll();
    std::cout << this->name << " rolls an attack bonus of " << temp << endl;
    temp2 = this->d20.roll();
    std::cout << opponent.getName() << " rolls a defense bonus of " << temp2 << endl;

    if((this->atk + temp) > (opponent.getDefense() + temp2)){
        std::cout << "The attack hits dealing 3-D6 damage!" << endl;

        temp = this->d6.roll();
        temp2 = this->d6.roll();
        temp3 = this->d6.roll();

        std::cout << "The rolls are " << temp << ", " << temp2 << ", and " << temp3 << " totaling: " << (temp + temp2 + temp3) << " damage!" << endl;
        opponent.setHP((opponent.getHP()-(temp + temp2 + temp3)));
        if(opponent.getHP() < 1){
            std::cout << opponent.getName() << " has been defeated!" << endl;
            return true;
        }
        else{
            std::cout << opponent.getName() << " has " << opponent.getHP() << " hit points left" << endl;
            return false;
        }
    }

    else{
        std::cout << "The attack missed!" << endl;
        return false;
    }
}
